pkg/boltz/ws.go

```diff
@@ -24,15 +25,16 @@ type SwapUpdate struct {
 }
 
 type Websocket struct {
-	Updates chan SwapUpdate
-
-	apiUrl        string
-	subscriptions chan bool
-	conn          *websocket.Conn
-	closed        bool
-	reconnect     bool
-	dialer        *websocket.Dialer
-	swapIds       []string
+	Updates     chan SwapUpdate
+	updatesLock sync.Mutex
+
+	apiUrl            string
+	subscriptions     chan bool
+	conn              *websocket.Conn
+	closed            bool
+	dialer            *websocket.Dialer
+	swapIds           []string
```

Comment:
Now that I think of it, shouldn't there be a lock on `swapIds` to handle concurrent writes?